Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp

```diff
@@ -607,19 +737,31 @@ void WebPrintOperationGtk::printDone(RefPtr<WebCore::FragmentedSharedBuffer>&& b
 
 #if USE(CAIRO)
 void WebPrintOperationGtk::print(cairo_surface_t* surface, double xDPI, double yDPI)
+#elif USE(SKIA)
+void WebPrintOperationGtk::print(sk_sp<SkDocument>& document, double xDPI, double yDPI)
+#endif
 {
     ASSERT(m_printContext);
+#if USE(SKIA)
+    ASSERT(document);
+#endif
 
     auto data = makeUnique<PrintPagesData>(this);
     if (!data->isValid) {
+#if USE(CAIRO)
         cairo_surface_finish(surface);
+#endif
         printDone(nullptr, invalidPageRangeToPrint(frameURL()));
         return;
     }
 
     m_xDPI = xDPI;
     m_yDPI = yDPI;
+#if USE(CAIRO)
     m_cairoContext = adoptRef(cairo_create(surface));
+#elif USE(SKIA)
+    m_skiaDocument = document;
```

Comment:
I suspect the memory management / refcounting is wrong here, but I can't prove how.

Reply:
I think it's ok, but this method could receive a `sk_sp<SkDocument>&&` instead and here you would `WTFMove(document)`